src/kili/gateways/kili_api_gateway/asset/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""Mixin extending Kili API Gateway class with Asset related operations."""
+
+
+from typing import Callable, List, Optional
+
+from kili.core.graphql.graphql_client import GraphQLClient
+from kili.gateways.kili_api_gateway.asset.operations import (
+    GQL_COUNT_ASSETS,
+    GQL_CREATE_UPLOAD_BUCKET_SIGNED_URLS,
+    get_asset_query,
+)
+from kili.gateways.kili_api_gateway.asset.types import AssetWhere
+from kili.gateways.kili_api_gateway.queries import (
+    PaginatedGraphQLQuery,
+    QueryOptions,
+    fragment_builder,
+    get_number_of_elements_to_query,
+)
+
+
+class AssetOperationMixin:
+    """Mixin extending Kili API Gateway class with Assets related operations."""
+
+    graphql_client: GraphQLClient
+
+    def list_assets(
+        self,
+        where: AssetWhere,
+        fields: List[str],
+        options: QueryOptions,
+        post_call_function: Optional[Callable],
+    ):
+        """List assets with given options."""
+        fragment = fragment_builder(fields)
+        query = get_asset_query(fragment)
+        nb_elements_to_query = get_number_of_elements_to_query(
+            self.graphql_client, where, options, GQL_COUNT_ASSETS
+        )
```

Comment:
is it needed? There is a `_count` in the PaginatedGraphQLQuery already

Reply:
ah well spotted,` _count` is a dead code and not called anywhere else. i delete it.

In order to make things easier and not to dependant, I moved the count query outside of the `PaginatedGraphQLQuery`. if one query has a count method and that it could provide a better user experience with `tqdm` logging, we can execute this query in the use case and pass it to the `execute_query_from_paginated_call` function for it to be aware of the number of elements it has to retrieve. If a query does not have an associated count query, then we can still paginate but without progress logging